src/Grpc.Net.Client/GrpcChannel.cs

```diff
@@ -221,7 +369,22 @@ private GrpcMethodInfo CreateMethodInfo(IMethod method)
             var scope = new GrpcCallScope(method.Type, uri);
             var methodConfig = ResolveMethodConfig(method);
 
-            return new GrpcMethodInfo(scope, new Uri(Address, uri), methodConfig);
+            var uriBuilder = new UriBuilder(Address);
+            uriBuilder.Path = method.FullName;
+
+            // The Uri used to create HttpRequestMessage must have a http or https scheme.
+            uriBuilder.Scheme = IsSecure ? Uri.UriSchemeHttps : Uri.UriSchemeHttp;
+
+            // A Uri with a http or https scheme requires a host name.
```

Comment:
nit: same issue with "authority" vs "host" terminology as mentioned above (currently the comments aren't untrue, so leaving the resolution up to you).